docs/configuration.rst

```diff
@@ -4,59 +4,197 @@
-configuration file will always take precedence over the global one.
+Where to configure shub
+-----------------------
 
-Both files have the same format::
+shub is configured via two YAML files:
+
+* ``~/.scrapinghub.yml`` -- this file contains global configuration like
+  your API key. It is automatically created in your home directory when you run
+  ``shub login``.
+* ``scrapinghub.yml`` -- this file contains local configuration like the
+  project ID or the location of your requirements file. It is automatically
+  created in your project directory when you run ``shub deploy`` for the first
+  time.
+
+All configuration options listed below can be used in both of these
+configuration files.  In case they overlap, the local configuration file will
+always take precedence over the global one.
+
+
+Defining target projects
+------------------------
+
+A very basic ``scrapinghub.yml``, as generated when you first run ``shub
+deploy``, could look like this::
+
+    project: 12345
+
+This tells shub to deploy to the Scrapy Cloud project ``12345`` when you run
+``shub deploy``.  Often, you will have multiple projects on Scrapy Cloud, e.g.
+one for development and one for production. For these cases, you can replace
+the ``project`` option with a ``projects`` dictionary::
 
     projects:
       default: 12345
       prod: 33333
 
-    # Populated manually or via shub login
-    apikey: 0bbf4f0f691e0d9378ae00ca7bcf7f0c
+shub will now deploy to project ``12345`` when you run ``shub deploy``, and
+deploy to project ``33333`` when you run ``shub deploy prod``.
+
+
+The configuration options
+-------------------------
+
+A deployed project contains more than your Scrapy code. Among other things, it
+has a version tag, and often has additional package requirements or is bound to
+a specific Scrapy version. All of these can be configured in
+``scrapinghub.yml``.
+
+Sometimes the requirements may be different for different target projects, e.g.
+because you want to run your development project on Scrapy 1.3 but use Scrapy
+1.0 for your production project. For these cases some options can be configured
+either *globally* or *project-specific*.
+
+A *global* configuration option serves as default for all projects. E.g., to
+set ``scrapy:1.3-py3`` as default `Scrapy Cloud stack`_, use::
+
+    projects:
+      default: 12345
+      prod: 33333
+
+    stack: scrapy:1.3-py3
+
+If you wish to use the stack *only* for project ``12345``, expand its entry in
+``projects`` as follows::
+
+    projects:
+      default:
+        id: 12345
+        stack: scrapy:1.3-py3
+      prod: 33333
+
+The following is a list of all available configuration options:
+
+================  ============================================  ===============
+Option            Description                                   Scope
+================  ============================================  ===============
+``requirements``  Path to the project's requirements file, and  global only
```

Comment:
Ah, sorry Viktor, I forgot about this comment, I didn't mean to ignore you. ;) I have ordered them by how often they are used (based on guessing). `requirements` is less complex than `stack`, which is less complex than `image`. `version` is used rarely as the default makes a lot of sense, and `apikey` is usually not configured by hand.

I think this ordering makes sense if new users are curious about what can be configured. On the other hand, if you're not a new user but want to use the table as a reference, it's harder to find the option you're looking for.